ext/ffi/jit_trampoline.rs

```diff
@@ -83,20 +86,35 @@ pub(crate) fn codegen(sym: &crate::Symbol) -> String {
     c += native_arg_to_c(ty);
     let _ = write!(c, " p{i}");
   }
+  if needs_unwrap {
+    let _ = write!(c, ", struct FastApiTypedArray* const p_ret");
```

Comment:
Maybe also change the `void* recv` to be `void* const recv` while we're here, since it similarly cannot be rewritten for any meaningful effect.

Reply:
I'm now curious as to whether the extra text and validation could have a negative impact on performance? Every "const " is an extra 2x6 bytes per function - that could add a minor amount of extra reallocations and copies for our Rust side, and some extra processing for the C compiler?

Reply:
These compilations are done once per symbol at `Deno.dlopen()` time, which is most likely done once. The `dlopen()` call isn't performance critical, most users probably wouldn't realise even if it took a whole second.

More impactful text removals would be eg. using `u8` etc. type names in the prelude (This would actually be more proper in some sense, as `_t` type names are reserved for official types to avoid name clashes. Of course, here we're just copying the types so it's not _really_ intruding.), using a shorter name for `FastApiTypedArray` etc. None of this is likely to make much of a performance difference.

Testing out compiling with and without `const`s on Godbolt doesn't show any meaningful difference, though there of course a majority of the time is probably just spent in the HTTP requests etc. Still, I doubt a few consts are going to make a meaningful difference in the `dlopen` performance, and we've already seen that not having them allowed for a surprising kind of (potential) bug to be introduced in the C code.

Reply:
Fair enough, I suppose that it's comparable to say `WebAssembly.instantiate` - rarely does one care about the initialisation time.